### PR TITLE
G-API: Fix bugs with dangling frames

### PR DESCRIPTION
**Pull request by TolyaTalamanov**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [ ] I agree to contribute to the project under Apache 2 License.
- [ ] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [ ] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### Build configuration
```
force_builders=Custom,Custom Win,Custom Mac
build_gapi_standalone:Linux x64=ade-0.1.1f
build_gapi_standalone:Win64=ade-0.1.1f
build_gapi_standalone:Mac=ade-0.1.1f
build_gapi_standalone:Linux x64 Debug=ade-0.1.1f

Xbuild_image:Custom=centos:7
Xbuildworker:Custom=linux-1
build_gapi_standalone:Custom=ade-0.1.1f

build_image:Custom=ubuntu-openvino-2021.3.0:20.04
build_image:Custom Win=openvino-2021.2.0
build_image:Custom Mac=openvino-2021.2.0

test_modules:Custom=gapi,python2,python3,java
test_modules:Custom Win=gapi,python2,python3,java
test_modules:Custom Mac=gapi,python2,python3,java

buildworker:Custom=linux-1
test_opencl:Custom=OFF
test_bigdata:Custom=1
test_filter:Custom=*
```

### CODE REVIEW

**Comment by dmatveev** [comment]
@TolyaTalamanov I believe this should be backported to 4.5.3-openvino as soon as we merge it - can you please pick this to that branch?

File: modules/gapi/src/executor/gstreamingexecutor.cpp [hunk 1]
```diff
@@ -347,6 +347,10 @@ bool QueueReader::getInputVector(std::vector<Q*> &in_queues,
                                  cv::GRunArgs    &in_constants,
                                  cv::GRunArgs    &isl_inputs)
 {
+    // NB: Need to release resources from the previous step, to fetch new ones.
+    // On some systems it might be impossible to allocate new memory
+    // until the old one is released.
+    m_cmd.clear();
```
Comment (dmatveev) [inline modules/gapi/src/executor/gstreamingexecutor.cpp:353]: what about the `isl_inputs` below?

**Review by dmatveev — COMMENTED** [review]
Fantastic!

File: modules/gapi/src/executor/gstreamingexecutor.cpp (hunk 1 above)
Comment (TolyaTalamanov) [reply modules/gapi/src/executor/gstreamingexecutor.cpp:353]: It's already empty: https://github.com/opencv/opencv/blob/master/modules/gapi/src/executor/gstreamingexecutor.cpp#L651-L653

**Comment by TolyaTalamanov** [comment]
> @TolyaTalamanov I believe this should be backported to 4.5.3-openvino as soon as we merge it - can you please pick this to that branch?

Of course, I'll pick it

**Review by dmatveev — APPROVED** [review]
👍

**Comment by TolyaTalamanov** [comment]
@alalek Could you merge it ?

**Pull request merged**